### DISCUSSION TITLE
Chia DB Upgrade

### DISCUSSION THREAD

**Discussion by Isanderthul** [guydavis/machinaris#660] [Q&A] [answered]
A few questions about the db upgrade:
1. Is the db upgrade process still `chia db upgrade` like in https://thechiaplot.net/2022/02/11/chia-database-upgrade-process-and-results/
2. How long does the upgrade take on a plain old ssd (would like to get a feel in terms of x hours)
3. can the docker farmer carry on while the upgrade is in progress?
4. Currently my daily db backup is 117GB, what would the size be after the upgrade?

**Comment by guydavis** [comment, marked as answer]
Greetings!  

1. Yes, just ` chia db upgrade` followed by `chia start farmer -r`, then delete old v1 files from /mnt/user/appdata/machinaris/mainnet/db folder.
2. Couple of hours on my old hardware.
3. Yes, the `chia db upgrade` command runs in an open  Unraid Console window, inside the Machinaris Container.  The Chia blockchain keeps farming using the v1 database while this happens.  Switch-over happens when you `chia start farmer -r`.
4. My v2 DB is about 65 GB in size, so make sure you have enough blank space in your appdata to hold both v1 and v2 databases at the same time during the conversion.

Hope this helps.

**Reply by Isanderthul** [reply]
```
-rw-r--r-- 1 root root 123G May  6 16:55 blockchain_v1_mainnet.sqlite
-rw-r--r-- 1 root root  35G May  6 01:53 blockchain_v2_mainnet.sqlite
```

**Comment by Isanderthul** [comment]
I did a `chia start farmer -r` but it seems to have gone back to using the v1 database
must I delete all the v1 databases after the switch? how do I ensure it stays on v2?

**Reply by Isanderthul** [reply]
@guydavis

**Reply by guydavis** [reply]
Hi, some have reported having to manually modify the Chia fullnode's config.yaml file as [outlined here](https://github.com/Chia-Network/chia-blockchain/wiki/FAQ#what-is-the-new-database).  If necessary, you can do this from the Settings | Farming page of the Machinaris WebUI.
